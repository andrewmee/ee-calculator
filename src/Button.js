import React from "react";

class Button extends React.Component {
  onClick = (e) => {
    e.preventDefault();
    this.props.onClick(this.props.value);
  };

  render() {
    let classNames = 'calculator__button';
    if (this.props.type === "operation") {
      classNames += ` calculator__button--operation calculator__button--${this.props.value}`
    }

    return (
      <button className={classNames} onClick={this.onClick} disabled={this.props.disabled}>
        {this.props.label || this.props.value}
      </button>
    );
  }
}

export default Button;
