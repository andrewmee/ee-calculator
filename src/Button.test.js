import React from 'react';
import { shallow } from 'enzyme';
import Button from './Button';

it('renders correctly', () => {
  const wrapper = shallow(<Button />);
  expect(wrapper).toMatchSnapshot();
});

it('renders correctly for operations', () => {
  const wrapper = shallow(<Button type="operation" value="foo" />);
  expect(wrapper).toMatchSnapshot();
});

it('handles the button click', () => {
  const spy = jest.fn();
  const wrapper = shallow(<Button value="foo" onClick={spy} />);
  wrapper.simulate('click', {
    preventDefault: () => {},
  });
  expect(spy).toHaveBeenCalledWith('foo');
});
