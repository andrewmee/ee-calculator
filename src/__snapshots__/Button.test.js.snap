// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<button
  className="calculator__button"
  onClick={[Function]}
/>
`;

exports[`renders correctly for operations 1`] = `
<button
  className="calculator__button calculator__button--operation calculator__button--foo"
  onClick={[Function]}
>
  foo
</button>
`;
